app/api.py

```diff
@@ -141,3 +157,54 @@ async def unsubscribe_to_track(track_namespace: str, subscriber: str):
                             detail="Relay is not in track namespace")
 
     track.remove_subscriber(subscriber)
+
+@app.get("/origin/{relayid}/{namespace}")
+async def get_origin(relayid: str, namespace: str):
+    nodes = network.nodes(data=True)
+    if len(nodes) < int(relayid):
+        raise HTTPException(status_code=response.status_code, detail="no such relay")
+
+    relay_in = None
+    for i, node in enumerate(nodes, start=1):
+        if (str(i) == relayid) & (relay_in == None):
+            relay_in = node[0]
+
+    response = await subscribe_to_track(track_namespace=namespace, subscriber=relay_in)
+    relay_out  = None
+    for i, node in enumerate(nodes, start=1):
+        if (node[0] == response) & (relay_out == None):
+            relay_out = i
```

Comment:
A másik irányhoz hasonlóan: `relay_out = list(nodes).index(response) + 1`.

De amúgy feel free to ignore me, mert tényleg nem lesz sokkal jobb (hatékonyságban).